ateam_common/include/ateam_common/overlay.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#ifndef ATEAM_COMMON__OVERLAY_HPP_
+#define ATEAM_COMMON__OVERLAY_HPP_
+
+#include <Eigen/Dense>
+
+#include <functional>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <ateam_msgs/msg/overlay.hpp>
+
+namespace ateam_common
+{
+class Overlay
+{
+public:
+  static Overlay & GetOverlay()
```

Comment:
Not a huge fan of this singleton helper, but not gonna block anything on this. Worth some future discussion.

Reply:
Wrote up ticket for it https://github.com/SSL-A-Team/software/issues/115